webapp/packages/plugin-settings-panel/src/SettingsPanel/getGroupsFromTree.ts

```diff
@@ -0,0 +1,21 @@
+/*
+ * CloudBeaver - Cloud Database Manager
+ * Copyright (C) 2020-2024 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0.
+ * you may not use this file except in compliance with the License.
+ */
+import { ROOT_SETTINGS_GROUP, type SettingsGroup as SettingsGroupType } from '@cloudbeaver/core-settings';
+import type { ITreeData } from '@cloudbeaver/plugin-navigation-tree';
+
+export function* getGroupsFromTree(treeData: ITreeData, groups: string[]): IterableIterator<SettingsGroupType> {
```

Comment:
What is the reason for the generator here?

Reply:
it's fix for this comment https://github.com/dbeaver/cloudbeaver/pull/2908#discussion_r1756768170